app/dependencies.py

```diff
@@ -86,9 +88,12 @@ def init_and_get_db_engine(settings: Settings) -> AsyncEngine:
         SQLALCHEMY_DATABASE_URL = f"postgresql+asyncpg://{settings.POSTGRES_USER}:{settings.POSTGRES_PASSWORD}@{settings.POSTGRES_HOST}/{settings.POSTGRES_DB}"
 
     if engine is None:
+        # The use of transactional database requires the use of NullPool as it implies the use of a db in multiple run loops
+        # https://docs.sqlalchemy.org/en/14/orm/extensions/asyncio.html#using-multiple-asyncio-event-loops
```

Comment:
We need to clear the usage of NullPool up. This should not break tests as this dependency should not be called during tests